util/chunk/chunk.go

```diff
@@ -706,3 +706,26 @@ func (c *Chunk) ToString(ft []*types.FieldType) string {
 	}
 	return string(buf)
 }
+
+// AppendRows appends multiple rows  to the chunk.
+func (c *Chunk) AppendRows(rows []Row) {
+	c.AppendPartialRows(rows)
+	c.numVirtualRows += len(rows)
+}
+
+// AppendPartialRows appends multiple row to the chunk.
+func (c *Chunk) AppendPartialRows(rows []Row) {
+	for i, dstCol := range c.columns {
+		for _, srcRow := range rows {
+			appendCellByCell(dstCol, srcRow.c.columns[i], srcRow.idx)
+		}
+	}
+}
+
+// LeftRequiredRows  chunk may append the number of rows
+func (c *Chunk) LeftRequiredRows(offset int) int {
+	if (c.requiredRows >= c.NumRows()+offset) || (c.requiredRows-c.NumRows() >= offset) {
```

Comment:
What the difference between the two conditions?

Reply:
> or <

Reply:
offset   > or <  requiredset